types/node/index.d.ts

```diff
@@ -4423,6 +4423,10 @@ declare module "path" {
         name: string;
     }
 
+    export interface FormatInputPathObject extends
+        Pick<ParsedPath, 'root' | 'dir' | 'ext' | 'name'>,
+        Partial<Pick<ParsedPath, 'base'>> {}
```

Comment:
I think all properties should be optional here.
* `root` is ignored if `dir` is provided
==> so why have root mandatory if it is ignored?
==> and why should we mandate to set `dir === ""` to get `root`used?
* `ext` and `name` are ignored if `base` exists
==> I see no reason for mandatory `ext` and `name` as they get ignored
==> not every path has an extension so why force users to set ""? 

Not sure if it is possible to define a type which represents these interdependency.

Reply:
> Not sure if it is possible to define a type which represents these interdependency.

I guess you could express a union of all the permutations:

``` ts
type Obj =
    { base: string; ext: string; name: string; }
  | { ext: string; name: string; } 
  | { base: string; };
```

However I think it's simpler just to make them all optional, like you said. After all, you can call `format` with an empty object and it will work (i.e. it won't throw, so it's type safe).

Reply:
@Flarna Updated to make all properties optional.